packages/client/src/runtime/core/protocol/graphql.ts

```diff
@@ -0,0 +1,138 @@
+
+  createMessage({ action, model, args, extensions, clientMethod, callsite }: CreateMessageOptions): ProtocolMessage {
+    let rootField: string | undefined
+    const operation = actionOperationMap[action]
+
+    if (action === 'executeRaw' || action === 'queryRaw' || action === 'runCommandRaw') {
+      rootField = action
+    }
+
+    let mapping
+    if (model !== undefined) {
+      mapping = this.dmmf?.mappingsMap[model]
+      if (mapping === undefined) {
+        throw new Error(`Could not find mapping for model ${model}`)
+      }
+
+      rootField = mapping[action === 'count' ? 'aggregate' : action]
+    }
+
+    if (operation !== 'query' && operation !== 'mutation') {
+      throw new Error(`Invalid operation ${operation} for action ${action}`)
+    }
+
+    const field = this.dmmf?.rootFieldMap[rootField!]
+
+    if (field === undefined) {
+      throw new Error(
+        `Could not find rootField ${rootField} for action ${action} for model ${model} on rootType ${operation}`,
+      )
+    }
+
+    // TODO: figure out if can be done in PrismaClient iteself
+    // const { isList } = field.outputType
+    // const typeName = getOutputTypeName(field.outputType.type)
+    // const rejectOnNotFound: RejectOnNotFound = getRejectOnNotFound(action, typeName, args, this._rejectOnNotFound)
+    // warnAboutRejectOnNotFound(rejectOnNotFound, jsModelName, action)
+
+    const document = makeDocument({
+      dmmf: this.dmmf,
+      rootField: rootField!,
+      rootTypeName: operation,
+      select: args,
+      modelName: model,
+      extensions: extensions,
+    })
+
+    document.validate(args, false, clientMethod, this.errorFormat, callsite)
+    return new GraphQLMessage(document)
+  }
+}
+
+export class GraphQLMessage implements ProtocolMessage {
+  constructor(private document: Document) {}
+
+  isWrite(): boolean {
+    return this.document.type === 'mutation'
+  }
+
+  getBatchId(): string | undefined {
+    if (!this.getRootField().startsWith('findUnique')) {
+      return undefined
+    }
+
+    // we generate a string for the fields we have used in the `where`
+    const args = this.document.children[0].args?.args
+      .map((a) => {
+        if (a.value instanceof Args) {
+          return `${a.key}-${a.value.args.map((a) => a.key).join(',')}`
+        }
+        return a.key
+      })
+      .join(',')
+
+    // we generate a string for the fields we have used in the `includes`
+    const selectionSet = this.document.children[0].children!.join(',')
+
+    // queries that share this token will be batched and collapsed altogether
+    return `${this.document.children[0].name}|${args}|${selectionSet}`
+  }
+
+  toDebugString(): string {
+    return String(this.document)
+  }
+
+  toEngineQuery(): EngineQuery {
+    return { query: String(this.document) }
+  }
+
+  deserializeResponse(data: unknown, dataPath: string[]): unknown {
+    const rootField = this.getRootField()
+
+    // TODO: figure out what it is
```

Comment:
This basically "lifts" the value of a result at a given path. And `select` and `include` are removed here for the fluent API. This is old code that I wanted to remove when I refactored the API to use proxies but never got to it.